components/Item/Item.js

```diff
@@ -79,17 +79,17 @@ const Item = props => (
       }
     `}</style>
     <li className="product_card">
-      <Link as={`/product/${props.name}`} href={`/product?name=${props.name}&image=${props.imageUrl}&price=${props.originalPrice}&dprice=${props.discountedPrice}`}>
+      <Link as={`/product/${props.id}`} href={`/product?id=${props.id}`}>
```

Comment:
Can we have `/product/${props.id}` and access it from `context.params` in getInitialProps? Cleaner URLs.